lib/ansible/module_utils/network/nso/nso.py

```diff
@@ -244,7 +244,7 @@ def _call(self, payload):
             raise NsoException(
                 'NSO returned HTTP code {0}, expected 200'.format(resp.status), {})
 
-        resp_body = resp.read()
+        resp_body = resp.read().decode('utf8')
```

Comment:
Instead of `.decode('utf8')` you should utilize `to_text` from `ansible.module_utils._text`